File: aidp-java-client/src/main/java/com/oracle/aidataplatform/dp/model/OntologyProjectStatus.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;



/**
 * Status history row for an ontology project, including publish lifecycle records.
**/
@jakarta.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=OntologyProjectStatus.Builder.class)

public final class OntologyProjectStatus  {
    @Deprecated
    @java.beans.ConstructorProperties({"statusId", "projectId", "projectVersion", "publishVersion", "status", "comment", "validationReport", "compiledArtifactRef", "errorMessage", "idempotencyKey", "timeCreated", "timeUpdated"})
    public OntologyProjectStatus(String statusId, String projectId, Integer projectVersion, Integer publishVersion, Status status, String comment, String validationReport, String compiledArtifactRef, String errorMessage, String idempotencyKey, java.util.Date timeCreated, java.util.Date timeUpdated) {
        super();
        this.statusId = statusId;
        this.projectId = projectId;
        this.projectVersion = projectVersion;
        this.publishVersion = publishVersion;
        this.status = status;
        this.comment = comment;
        this.validationReport = validationReport;
        this.compiledArtifactRef = compiledArtifactRef;
        this.errorMessage = errorMessage;
        this.idempotencyKey = idempotencyKey;
        this.timeCreated = timeCreated;
        this.timeUpdated = timeUpdated;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
            
@com.fasterxml.jackson.annotation.JsonProperty("statusId")
private String statusId;



public Builder statusId(String statusId) {
    this.statusId = statusId;
    return this;
}
        
@com.fasterxml.jackson.annotation.JsonProperty("projectId")
private String projectId;



public Builder projectId(String projectId) {
    this.projectId = projectId;
    return this;
}
            /**
     * OMS project metadata version captured when publish starts.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("projectVersion")
private Integer projectVersion;

        /**
         * OMS project metadata version captured when publish starts.
         * @param projectVersion the value to set
         * @return this builder
         **/
        

public Builder projectVersion(Integer projectVersion) {
    this.projectVersion = projectVersion;
    return this;
}
            /**
     * Monotonic publish version for the project.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("publishVersion")
private Integer publishVersion;

        /**
         * Monotonic publish version for the project.
         * @param publishVersion the value to set
         * @return this builder
         **/
        

public Builder publishVersion(Integer publishVersion) {
    this.publishVersion = publishVersion;
    return this;
}
        
@com.fasterxml.jackson.annotation.JsonProperty("status")
private Status status;



public Builder status(Status status) {
    this.status = status;
    return this;
}
        
@com.fasterxml.jackson.annotation.JsonProperty("comment")
private String comment;



public Builder comment(String comment) {
    this.comment = comment;
    return this;
}
            /**
     * JSON validation report produced by a compiler worker.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("validationReport")
private String validationReport;

        /**
         * JSON validation report produced by a compiler worker.
         * @param validationReport the value to set
         * @return this builder
         **/
        

public Builder validationReport(String validationReport) {
    this.validationReport = validationReport;
    return this;
}
        
@com.fasterxml.jackson.annotation.JsonProperty("compiledArtifactRef")
private String compiledArtifactRef;



public Builder compiledArtifactRef(String compiledArtifactRef) {
    this.compiledArtifactRef = compiledArtifactRef;
    return this;
}
        
@com.fasterxml.jackson.annotation.JsonProperty("errorMessage")
private String errorMessage;



public Builder errorMessage(String errorMessage) {
    this.errorMessage = errorMessage;
    return this;
}
        
@com.fasterxml.jackson.annotation.JsonProperty("idempotencyKey")
private String idempotencyKey;



public Builder idempotencyKey(String idempotencyKey) {
    this.idempotencyKey = idempotencyKey;
    return this;
}
        
@com.fasterxml.jackson.annotation.JsonProperty("timeCreated")
private java.util.Date timeCreated;



public Builder timeCreated(java.util.Date timeCreated) {
    this.timeCreated = timeCreated;
    return this;
}
        
@com.fasterxml.jackson.annotation.JsonProperty("timeUpdated")
private java.util.Date timeUpdated;



public Builder timeUpdated(java.util.Date timeUpdated) {
    this.timeUpdated = timeUpdated;
    return this;
}


        public OntologyProjectStatus build() {
            OntologyProjectStatus model = new OntologyProjectStatus(this.statusId
                , this.projectId
                , this.projectVersion
                , this.publishVersion
                , this.status
                , this.comment
                , this.validationReport
                , this.compiledArtifactRef
                , this.errorMessage
                , this.idempotencyKey
                , this.timeCreated
                , this.timeUpdated);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(OntologyProjectStatus model) {
                this.statusId(model.getStatusId());
    this.projectId(model.getProjectId());
    this.projectVersion(model.getProjectVersion());
    this.publishVersion(model.getPublishVersion());
    this.status(model.getStatus());
    this.comment(model.getComment());
    this.validationReport(model.getValidationReport());
    this.compiledArtifactRef(model.getCompiledArtifactRef());
    this.errorMessage(model.getErrorMessage());
    this.idempotencyKey(model.getIdempotencyKey());
    this.timeCreated(model.getTimeCreated());
    this.timeUpdated(model.getTimeUpdated());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


    
    @com.fasterxml.jackson.annotation.JsonProperty("statusId")
    private final String statusId;

    
    public String getStatusId() {
        return statusId;
    }


    
    @com.fasterxml.jackson.annotation.JsonProperty("projectId")
    private final String projectId;

    
    public String getProjectId() {
        return projectId;
    }


        /**
     * OMS project metadata version captured when publish starts.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("projectVersion")
    private final Integer projectVersion;

        /**
     * OMS project metadata version captured when publish starts.
     * @return the value
     **/
    
    public Integer getProjectVersion() {
        return projectVersion;
    }


        /**
     * Monotonic publish version for the project.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("publishVersion")
    private final Integer publishVersion;

        /**
     * Monotonic publish version for the project.
     * @return the value
     **/
    
    public Integer getPublishVersion() {
        return publishVersion;
    }

    /**
     **/
    public enum Status implements com.oracle.bmc.http.internal.BmcEnum {
        Created("CREATED"),
        Updated("UPDATED"),
        Validating("VALIDATING"),
        ValidationFailed("VALIDATION_FAILED"),
        Publishing("PUBLISHING"),
        Published("PUBLISHED"),
        PublishFailed("PUBLISH_FAILED"),
        Unpublishing("UNPUBLISHING"),
        Unpublished("UNPUBLISHED"),
        UnpublishFailed("UNPUBLISH_FAILED"),
        Archived("ARCHIVED"),
        ;

        

        private final String value;
        private static java.util.Map<String, Status> map;

        static {
            map = new java.util.HashMap<>();
            for (Status v : Status.values()) {
                    map.put(v.getValue(), v);
                
            }
        }

        Status(String value) {
            this.value = value;
        }

        @com.fasterxml.jackson.annotation.JsonValue
        public String getValue() {
            return value;
        }

        @com.fasterxml.jackson.annotation.JsonCreator
        public static Status create(String key) {
            if (map.containsKey(key)) {
                return map.get(key);
            }
            throw new IllegalArgumentException("Invalid Status: " + key);
        }
    };
    
    @com.fasterxml.jackson.annotation.JsonProperty("status")
    private final Status status;

    
    public Status getStatus() {
        return status;
    }


    
    @com.fasterxml.jackson.annotation.JsonProperty("comment")
    private final String comment;

    
    public String getComment() {
        return comment;
    }


        /**
     * JSON validation report produced by a compiler worker.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("validationReport")
    private final String validationReport;

        /**
     * JSON validation report produced by a compiler worker.
     * @return the value
     **/
    
    public String getValidationReport() {
        return validationReport;
    }


    
    @com.fasterxml.jackson.annotation.JsonProperty("compiledArtifactRef")
    private final String compiledArtifactRef;

    
    public String getCompiledArtifactRef() {
        return compiledArtifactRef;
    }


    
    @com.fasterxml.jackson.annotation.JsonProperty("errorMessage")
    private final String errorMessage;

    
    public String getErrorMessage() {
        return errorMessage;
    }


    
    @com.fasterxml.jackson.annotation.JsonProperty("idempotencyKey")
    private final String idempotencyKey;

    
    public String getIdempotencyKey() {
        return idempotencyKey;
    }


    
    @com.fasterxml.jackson.annotation.JsonProperty("timeCreated")
    private final java.util.Date timeCreated;

    
    public java.util.Date getTimeCreated() {
        return timeCreated;
    }


    
    @com.fasterxml.jackson.annotation.JsonProperty("timeUpdated")
    private final java.util.Date timeUpdated;

    
    public java.util.Date getTimeUpdated() {
        return timeUpdated;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("OntologyProjectStatus(");
        sb.append("statusId=").append(String.valueOf(this.statusId));
        sb.append(", projectId=").append(String.valueOf(this.projectId));
        sb.append(", projectVersion=").append(String.valueOf(this.projectVersion));
        sb.append(", publishVersion=").append(String.valueOf(this.publishVersion));
        sb.append(", status=").append(String.valueOf(this.status));
        sb.append(", comment=").append(String.valueOf(this.comment));
        sb.append(", validationReport=").append(String.valueOf(this.validationReport));
        sb.append(", compiledArtifactRef=").append(String.valueOf(this.compiledArtifactRef));
        sb.append(", errorMessage=").append(String.valueOf(this.errorMessage));
        sb.append(", idempotencyKey=").append(String.valueOf(this.idempotencyKey));
        sb.append(", timeCreated=").append(String.valueOf(this.timeCreated));
        sb.append(", timeUpdated=").append(String.valueOf(this.timeUpdated));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof OntologyProjectStatus)) {
            return false;
        }

        OntologyProjectStatus other = (OntologyProjectStatus) o;
        return java.util.Objects.equals(this.statusId, other.statusId) &&
            java.util.Objects.equals(this.projectId, other.projectId) &&
            java.util.Objects.equals(this.projectVersion, other.projectVersion) &&
            java.util.Objects.equals(this.publishVersion, other.publishVersion) &&
            java.util.Objects.equals(this.status, other.status) &&
            java.util.Objects.equals(this.comment, other.comment) &&
            java.util.Objects.equals(this.validationReport, other.validationReport) &&
            java.util.Objects.equals(this.compiledArtifactRef, other.compiledArtifactRef) &&
            java.util.Objects.equals(this.errorMessage, other.errorMessage) &&
            java.util.Objects.equals(this.idempotencyKey, other.idempotencyKey) &&
            java.util.Objects.equals(this.timeCreated, other.timeCreated) &&
            java.util.Objects.equals(this.timeUpdated, other.timeUpdated);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.statusId == null ? 43 : this.statusId.hashCode());
        result = (result * PRIME) + (this.projectId == null ? 43 : this.projectId.hashCode());
        result = (result * PRIME) + (this.projectVersion == null ? 43 : this.projectVersion.hashCode());
        result = (result * PRIME) + (this.publishVersion == null ? 43 : this.publishVersion.hashCode());
        result = (result * PRIME) + (this.status == null ? 43 : this.status.hashCode());
        result = (result * PRIME) + (this.comment == null ? 43 : this.comment.hashCode());
        result = (result * PRIME) + (this.validationReport == null ? 43 : this.validationReport.hashCode());
        result = (result * PRIME) + (this.compiledArtifactRef == null ? 43 : this.compiledArtifactRef.hashCode());
        result = (result * PRIME) + (this.errorMessage == null ? 43 : this.errorMessage.hashCode());
        result = (result * PRIME) + (this.idempotencyKey == null ? 43 : this.idempotencyKey.hashCode());
        result = (result * PRIME) + (this.timeCreated == null ? 43 : this.timeCreated.hashCode());
        result = (result * PRIME) + (this.timeUpdated == null ? 43 : this.timeUpdated.hashCode());
        return result;
    }


}

File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/ontology_project_status.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.



from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class OntologyProjectStatus(object):
    """
    Status history row for an ontology project, including publish lifecycle records.
    """

    #: A constant which can be used with the status property of a OntologyProjectStatus.
    #: This constant has a value of "CREATED"
    STATUS_CREATED = "CREATED"

    #: A constant which can be used with the status property of a OntologyProjectStatus.
    #: This constant has a value of "UPDATED"
    STATUS_UPDATED = "UPDATED"

    #: A constant which can be used with the status property of a OntologyProjectStatus.
    #: This constant has a value of "VALIDATING"
    STATUS_VALIDATING = "VALIDATING"

    #: A constant which can be used with the status property of a OntologyProjectStatus.
    #: This constant has a value of "VALIDATION_FAILED"
    STATUS_VALIDATION_FAILED = "VALIDATION_FAILED"

    #: A constant which can be used with the status property of a OntologyProjectStatus.
    #: This constant has a value of "PUBLISHING"
    STATUS_PUBLISHING = "PUBLISHING"

    #: A constant which can be used with the status property of a OntologyProjectStatus.
    #: This constant has a value of "PUBLISHED"
    STATUS_PUBLISHED = "PUBLISHED"

    #: A constant which can be used with the status property of a OntologyProjectStatus.
    #: This constant has a value of "PUBLISH_FAILED"
    STATUS_PUBLISH_FAILED = "PUBLISH_FAILED"

    #: A constant which can be used with the status property of a OntologyProjectStatus.
    #: This constant has a value of "UNPUBLISHING"
    STATUS_UNPUBLISHING = "UNPUBLISHING"

    #: A constant which can be used with the status property of a OntologyProjectStatus.
    #: This constant has a value of "UNPUBLISHED"
    STATUS_UNPUBLISHED = "UNPUBLISHED"

    #: A constant which can be used with the status property of a OntologyProjectStatus.
    #: This constant has a value of "UNPUBLISH_FAILED"
    STATUS_UNPUBLISH_FAILED = "UNPUBLISH_FAILED"

    #: A constant which can be used with the status property of a OntologyProjectStatus.
    #: This constant has a value of "ARCHIVED"
    STATUS_ARCHIVED = "ARCHIVED"

    def __init__(self, **kwargs):
        """
        Initializes a new OntologyProjectStatus object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param status_id:
            The value to assign to the status_id property of this OntologyProjectStatus.
        :type status_id: str

        :param project_id:
            The value to assign to the project_id property of this OntologyProjectStatus.
        :type project_id: str

        :param project_version:
            The value to assign to the project_version property of this OntologyProjectStatus.
        :type project_version: int

        :param publish_version:
            The value to assign to the publish_version property of this OntologyProjectStatus.
        :type publish_version: int

        :param status:
            The value to assign to the status property of this OntologyProjectStatus.
            Allowed values for this property are: "CREATED", "UPDATED", "VALIDATING", "VALIDATION_FAILED", "PUBLISHING", "PUBLISHED", "PUBLISH_FAILED", "UNPUBLISHING", "UNPUBLISHED", "UNPUBLISH_FAILED", "ARCHIVED"
        :type status: str

        :param comment:
            The value to assign to the comment property of this OntologyProjectStatus.
        :type comment: str

        :param validation_report:
            The value to assign to the validation_report property of this OntologyProjectStatus.
        :type validation_report: str

        :param compiled_artifact_ref:
            The value to assign to the compiled_artifact_ref property of this OntologyProjectStatus.
        :type compiled_artifact_ref: str

        :param error_message:
            The value to assign to the error_message property of this OntologyProjectStatus.
        :type error_message: str

        :param idempotency_key:
            The value to assign to the idempotency_key property of this OntologyProjectStatus.
        :type idempotency_key: str

        :param time_created:
            The value to assign to the time_created property of this OntologyProjectStatus.
        :type time_created: datetime

        :param time_updated:
            The value to assign to the time_updated property of this OntologyProjectStatus.
        :type time_updated: datetime

        """
        self.swagger_types = {
            'status_id': 'str',
            'project_id': 'str',
            'project_version': 'int',
            'publish_version': 'int',
            'status': 'str',
            'comment': 'str',
            'validation_report': 'str',
            'compiled_artifact_ref': 'str',
            'error_message': 'str',
            'idempotency_key': 'str',
            'time_created': 'datetime',
            'time_updated': 'datetime'
        }

        self.attribute_map = {
            'status_id': 'statusId',
            'project_id': 'projectId',
            'project_version': 'projectVersion',
            'publish_version': 'publishVersion',
            'status': 'status',
            'comment': 'comment',
            'validation_report': 'validationReport',
            'compiled_artifact_ref': 'compiledArtifactRef',
            'error_message': 'errorMessage',
            'idempotency_key': 'idempotencyKey',
            'time_created': 'timeCreated',
            'time_updated': 'timeUpdated'
        }

        self._status_id = None
        self._project_id = None
        self._project_version = None
        self._publish_version = None
        self._status = None
        self._comment = None
        self._validation_report = None
        self._compiled_artifact_ref = None
        self._error_message = None
        self._idempotency_key = None
        self._time_created = None
        self._time_updated = None

    @property
    def status_id(self):
        """
        **[Required]** Gets the status_id of this OntologyProjectStatus.

        :return: The status_id of this OntologyProjectStatus.
        :rtype: str
        """
        return self._status_id

    @status_id.setter
    def status_id(self, status_id):
        """
        Sets the status_id of this OntologyProjectStatus.

        :param status_id: The status_id of this OntologyProjectStatus.
        :type: str
        """
        self._status_id = status_id

    @property
    def project_id(self):
        """
        **[Required]** Gets the project_id of this OntologyProjectStatus.

        :return: The project_id of this OntologyProjectStatus.
        :rtype: str
        """
        return self._project_id

    @project_id.setter
    def project_id(self, project_id):
        """
        Sets the project_id of this OntologyProjectStatus.

        :param project_id: The project_id of this OntologyProjectStatus.
        :type: str
        """
        self._project_id = project_id

    @property
    def project_version(self):
        """
        **[Required]** Gets the project_version of this OntologyProjectStatus.
        OMS project metadata version captured when publish starts.


        :return: The project_version of this OntologyProjectStatus.
        :rtype: int
        """
        return self._project_version

    @project_version.setter
    def project_version(self, project_version):
        """
        Sets the project_version of this OntologyProjectStatus.
        OMS project metadata version captured when publish starts.


        :param project_version: The project_version of this OntologyProjectStatus.
        :type: int
        """
        self._project_version = project_version

    @property
    def publish_version(self):
        """
        Gets the publish_version of this OntologyProjectStatus.
        Monotonic publish version for the project.


        :return: The publish_version of this OntologyProjectStatus.
        :rtype: int
        """
        return self._publish_version

    @publish_version.setter
    def publish_version(self, publish_version):
        """
        Sets the publish_version of this OntologyProjectStatus.
        Monotonic publish version for the project.


        :param publish_version: The publish_version of this OntologyProjectStatus.
        :type: int
        """
        self._publish_version = publish_version

    @property
    def status(self):
        """
        **[Required]** Gets the status of this OntologyProjectStatus.
        Allowed values for this property are: "CREATED", "UPDATED", "VALIDATING", "VALIDATION_FAILED", "PUBLISHING", "PUBLISHED", "PUBLISH_FAILED", "UNPUBLISHING", "UNPUBLISHED", "UNPUBLISH_FAILED", "ARCHIVED"


        :return: The status of this OntologyProjectStatus.
        :rtype: str
        """
        return self._status

    @status.setter
    def status(self, status):
        """
        Sets the status of this OntologyProjectStatus.

        :param status: The status of this OntologyProjectStatus.
        :type: str
        """
        allowed_values = ["CREATED", "UPDATED", "VALIDATING", "VALIDATION_FAILED", "PUBLISHING", "PUBLISHED", "PUBLISH_FAILED", "UNPUBLISHING", "UNPUBLISHED", "UNPUBLISH_FAILED", "ARCHIVED"]
        if not value_allowed_none_or_none_sentinel(status, allowed_values):
            raise ValueError(
                "Invalid value for `status`, must be None or one of {0}"
                .format(allowed_values)
            )
        self._status = status

    @property
    def comment(self):
        """
        Gets the comment of this OntologyProjectStatus.

        :return: The comment of this OntologyProjectStatus.
        :rtype: str
        """
        return self._comment

    @comment.setter
    def comment(self, comment):
        """
        Sets the comment of this OntologyProjectStatus.

        :param comment: The comment of this OntologyProjectStatus.
        :type: str
        """
        self._comment = comment

    @property
    def validation_report(self):
        """
        Gets the validation_report of this OntologyProjectStatus.
        JSON validation report produced by a compiler worker.


        :return: The validation_report of this OntologyProjectStatus.
        :rtype: str
        """
        return self._validation_report

    @validation_report.setter
    def validation_report(self, validation_report):
        """
        Sets the validation_report of this OntologyProjectStatus.
        JSON validation report produced by a compiler worker.


        :param validation_report: The validation_report of this OntologyProjectStatus.
        :type: str
        """
        self._validation_report = validation_report

    @property
    def compiled_artifact_ref(self):
        """
        Gets the compiled_artifact_ref of this OntologyProjectStatus.

        :return: The compiled_artifact_ref of this OntologyProjectStatus.
        :rtype: str
        """
        return self._compiled_artifact_ref

    @compiled_artifact_ref.setter
    def compiled_artifact_ref(self, compiled_artifact_ref):
        """
        Sets the compiled_artifact_ref of this OntologyProjectStatus.

        :param compiled_artifact_ref: The compiled_artifact_ref of this OntologyProjectStatus.
        :type: str
        """
        self._compiled_artifact_ref = compiled_artifact_ref

    @property
    def error_message(self):
        """
        Gets the error_message of this OntologyProjectStatus.

        :return: The error_message of this OntologyProjectStatus.
        :rtype: str
        """
        return self._error_message

    @error_message.setter
    def error_message(self, error_message):
        """
        Sets the error_message of this OntologyProjectStatus.

        :param error_message: The error_message of this OntologyProjectStatus.
        :type: str
        """
        self._error_message = error_message

    @property
    def idempotency_key(self):
        """
        Gets the idempotency_key of this OntologyProjectStatus.

        :return: The idempotency_key of this OntologyProjectStatus.
        :rtype: str
        """
        return self._idempotency_key

    @idempotency_key.setter
    def idempotency_key(self, idempotency_key):
        """
        Sets the idempotency_key of this OntologyProjectStatus.

        :param idempotency_key: The idempotency_key of this OntologyProjectStatus.
        :type: str
        """
        self._idempotency_key = idempotency_key

    @property
    def time_created(self):
        """
        Gets the time_created of this OntologyProjectStatus.

        :return: The time_created of this OntologyProjectStatus.
        :rtype: datetime
        """
        return self._time_created

    @time_created.setter
    def time_created(self, time_created):
        """
        Sets the time_created of this OntologyProjectStatus.

        :param time_created: The time_created of this OntologyProjectStatus.
        :type: datetime
        """
        self._time_created = time_created

    @property
    def time_updated(self):
        """
        Gets the time_updated of this OntologyProjectStatus.

        :return: The time_updated of this OntologyProjectStatus.
        :rtype: datetime
        """
        return self._time_updated

    @time_updated.setter
    def time_updated(self, time_updated):
        """
        Sets the time_updated of this OntologyProjectStatus.

        :param time_updated: The time_updated of this OntologyProjectStatus.
        :type: datetime
        """
        self._time_updated = time_updated

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other
